src/bam/record.rs

```diff
@@ -666,6 +668,64 @@ impl CigarString {
     pub fn into_view(self, pos: i32) -> CigarStringView {
         CigarStringView::new(self, pos)
     }
+
+    /// Calculate the bam cigar from the alignment struct. x is the target string
+    /// and y is the reference.
+    #[cfg(feature = "bio")]
+    pub fn from_alignment(alignment: &Alignment, hard_clip: bool) -> Self {
```

Comment:
Maybe implement the `From` trait instead (see [here](https://doc.rust-lang.org/std/convert/trait.From.html))?